File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    let answer = [];
    const keytable = keymap.reduce((obj, keys) => {
        [...keys].forEach((key, i) => {
            if (obj[key]) {
                obj[key] = Math.min(obj[key], i + 1)
            } else {
                obj[key] = i + 1
            }
        })
        return obj
      }, {})
    
    answer = targets.map((target) => {
        let clicked = 0
        for (const key of [...target]) {
            
            if (!keytable[key]) {
                clicked = -1
                break
            }
            clicked += keytable[key]
        }
        return clicked
    })
    return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = [0] * len(targets)
    
    temp = dict()
    for key in keymap:
        for idx,k in enumerate(key):
            if k in temp:
                temp[k] = min(temp[k],idx+1)
            else:
                temp[k] = idx + 1
    
    print(temp)
    
    for idx,tar in enumerate(targets):
        
        for t in tar:
            if t in temp:
                answer[idx] += temp[t]
            else:
                answer[idx] = -1
                break
    return answer